File: mobile-java/src/com/meb/admin/PinManager.java
package com.meb.admin;

import android.content.Context;
import android.content.SharedPreferences;
import java.security.MessageDigest;

public class PinManager {

    private static final String PREFS = "meb_secure";

    public static boolean isPinSet(Context ctx) {
        return prefs(ctx).getString("pin_hash", null) != null;
    }

    public static void setPin(Context ctx, String pin) {
        prefs(ctx).edit().putString("pin_hash", sha(pin)).apply();
    }

    public static boolean checkPin(Context ctx, String pin) {
        String stored = prefs(ctx).getString("pin_hash", null);
        return stored != null && stored.equals(sha(pin));
    }

    public static void clearPin(Context ctx) {
        prefs(ctx).edit().remove("pin_hash").apply();
    }

    private static SharedPreferences prefs(Context ctx) {
        return ctx.getSharedPreferences(PREFS, Context.MODE_PRIVATE);
    }

    private static String sha(String s) {
        try {
            MessageDigest md = MessageDigest.getInstance("SHA-256");
            byte[] digest = md.digest(s.getBytes("UTF-8"));
            StringBuilder sb = new StringBuilder();
            for (byte b : digest) {
                sb.append(String.format("%02x", b));
            }
            return sb.toString();
        } catch (Exception e) {
            return s;
        }
    }
}

File: mobile/app/src/main/java/com/meb/admin/PinManager.kt
package com.meb.admin

import android.content.Context
import java.security.MessageDigest

object PinManager {
    private const val PREFS = "meb_secure"
    private fun prefs(ctx: Context) = ctx.getSharedPreferences(PREFS, Context.MODE_PRIVATE)

    fun isPinSet(ctx: Context) = prefs(ctx).getString("pin_hash", null) != null

    fun setPin(ctx: Context, pin: String) =
        prefs(ctx).edit().putString("pin_hash", sha(pin)).apply()

    fun checkPin(ctx: Context, pin: String): Boolean =
        prefs(ctx).getString("pin_hash", null) == sha(pin)

    fun clear(ctx: Context) = prefs(ctx).edit().remove("pin_hash").apply()

    private fun sha(s: String): String =
        MessageDigest.getInstance("SHA-256").digest(s.toByteArray())
            .joinToString("") { "%02x".format(it) }

    fun cacheLeads(ctx: Context, json: String) {
        ctx.openFileOutput("leads_cache.json", Context.MODE_PRIVATE).use { it.write(json.toByteArray()) }
    }

    fun loadCachedLeads(ctx: Context): String? = try {
        ctx.openFileInput("leads_cache.json").bufferedReader().use { it.readText() }
    } catch (e: Exception) { null }
}
